plugins/inputs/nvidia_smi/nvidia_smi.go

```diff
@@ -119,8 +126,10 @@ func (smi *NvidiaSMI) parse(acc telegraf.Accumulator, data []byte) error {
 func init() {
 	inputs.Add("nvidia_smi", func() telegraf.Input {
 		return &NvidiaSMI{
-			BinPath: "/usr/bin/nvidia-smi",
-			Timeout: config.Duration(5 * time.Second),
+			BinPath:       "/usr/bin/nvidia-smi",
+			Timeout:       config.Duration(5 * time.Second),
+			TestOnStartup: false,
+			nvidiaSMIArgs: []string{"-q", "-x"},
```

Comment:
Note: the reason I'm breaking this out is because I want to modify the arguments used in testing to force a non-zero exit code. The tests themselves don't appear to actually integrate with a real nvidia-smi executable, so the solution I came up with is to just run `/bin/bash -c "exit 9"` as an example of the command returning non-zero.